File: src/main/java/com/freya02/bot/docs/cached/CachedMethod.java
package com.freya02.bot.docs.cached;

import com.freya02.docs.SeeAlso;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.List;

public final class CachedMethod implements CachedDoc {
	private final MessageEmbed methodEmbed;
	private final List<SeeAlso.SeeAlsoReference> seeAlsoReferences;

	public CachedMethod(MessageEmbed methodEmbed,
	                    List<SeeAlso.SeeAlsoReference> seeAlsoReferences) {
		this.methodEmbed = methodEmbed;
		this.seeAlsoReferences = seeAlsoReferences;
	}

	public MessageEmbed getMethodEmbed() {
		return methodEmbed;
	}

	@Override
	public List<SeeAlso.SeeAlsoReference> getSeeAlsoReferences() {
		return seeAlsoReferences;
	}
}

File: src/main/kotlin/com/freya02/bot/docs/cached/CachedMethod.kt
package com.freya02.bot.docs.cached

import com.freya02.bot.docs.metadata.ImplementationIndex
import com.freya02.docs.DocSourceType
import com.freya02.docs.data.SeeAlso.SeeAlsoReference
import net.dv8tion.jda.api.entities.MessageEmbed

class CachedMethod(
    override val source: DocSourceType,
    val className: String,
    val signature: String,
    override val embed: MessageEmbed,
    override val seeAlsoReferences: List<SeeAlsoReference>,
    override val javadocLink: String?,
    override val sourceLink: String?,
    val implementations: List<ImplementationIndex.Method>,
    val overriddenMethods: List<ImplementationIndex.Method>
) : CachedDoc {
    override val qualifiedName: String
        get() = "$className#$signature"
}